app/assets/javascripts/app/datepicker-validate.coffee

```diff
@@ -0,0 +1,27 @@
+$ ->
+  # Parse to a date if it's a string, otherwise just return the original argument.
+  # This will handle "5/6/2016", unix timestamps, and `Date`s.
+  parseDate = (dateOrString, format) ->
+    if typeof dateOrString is "string"
+      $.datepicker.parseDate(format, dateOrString)
+    else
+      dateOrString
+
+  $(".datepicker__data").on "change", (e) ->
+    $input = $(e.target)
+
+    format = $input.datepicker("option", "dateFormat")
+    minDate = parseDate($input.datepicker("option", "minDate"), format)
+    maxDate = parseDate($input.datepicker("option", "maxDate"), format)
+
+    try
+      date = parseDate($input.val(), format)
+
+      if maxDate && date > maxDate
+        throw new Error("Beyond maxDate")
+      if minDate && date < minDate
+        throw new Error("Before minDate")
+
+      $input.closest(".control-group").removeClass("error")
+    catch e
+      $input.closest(".control-group").addClass("error")
```

Comment:
I would pull some or all of this into its own method or class, but then I'm generally kind of an outlier in JS code structure (I normally like event handlers to call regular functions or methods).